src/axom/inlet/examples/document_generation.cpp

```diff
@@ -0,0 +1,138 @@
+// usage : ./document_generation_example --enableDocs --deck lua_file 
+
+#include "axom/inlet/SphinxDocWriter.hpp"
+#include "axom/inlet/Inlet.hpp"
+#include "axom/inlet/LuaReader.hpp"
+
+#include "CLI11/CLI11.hpp"
+#include <iostream>
+
+using axom::inlet::Inlet;
+using axom::inlet::LuaReader;
+using axom::sidre::DataStore;
+using axom::inlet::SphinxDocWriter;
+
+void findStr(std::string path, std::shared_ptr<Inlet> inlet) {
+  std::string strVal;
+  bool found = inlet->get(path, strVal);
+  std::cout << path << ": ";
+  found ? std::cout << "found " << strVal : std::cout << "not found ";
+  std::cout << std::endl;
+} 
+
+void findInt(std::string path, std::shared_ptr<Inlet> inlet) {
+  int val;
+  bool found = inlet->get(path, val);
+  std::cout << path << ": ";
+  found ? std::cout << "found " << val : std::cout << "not found ";
+  std::cout << std::endl;
+} 
+
+void findDouble(std::string path, std::shared_ptr<Inlet> inlet) {
+  double val;
+  bool found = inlet->get(path, val);
+  std::cout << path << ": ";
+  found ? std::cout << "found " << val : std::cout << "not found ";
+  std::cout << std::endl;
+} 
+
+void exampleInlet(DataStore* ds, const std::string& luaFile, bool docsEnabled)
+{
+  auto lr = std::make_shared<LuaReader>();
+  lr->parseFile(luaFile);
+
+  auto inlet = std::make_shared<Inlet>(lr, ds->getRoot(), docsEnabled);
+
+  auto docWriter = std::make_shared<SphinxDocWriter>("example_docs.rst", inlet->sidreGroup());
+  inlet->registerDocWriter(docWriter);
+  
+  std::shared_ptr<axom::inlet::Field> currField;
+
+  currField = inlet->addString("thermal_solver/mesh/filename", "file for thermal solver");

```

Comment:
It would probably be good to have some fields that do not supply a description in this example.

Reply:
Looks like this may not be possible because the add functions all require descriptions, and without using an add function the field would not be found.

Reply:
You could change the add functions to have a default argument for description.  I meant to do that at some point and it got lost in one of my many rewrites.

Reply:
Should the default description just be an empty str?